vulcanai2/models/cnn.py

```diff
@@ -92,8 +94,42 @@ def __init__(self, name, in_dim, config, save_path=None,
             lr_scheduler, early_stopping, criter_spec)
 
     def _create_network(self, **kwargs):
-
+        self._in_dim = self.in_dim
         conv_hid_layers = self._config.units
+
+        if len(self.in_dim) > 1:
+            # Calculate converged in_dim for the MultiInput ConvNet
+            dim_tmp = []
+            # Create empty input tensors
+            in_tensors = [torch.empty(x) for x in self.in_dim]
+            # Calculate the number of elements in each input tensor 
+            els = [t.numel() for t in in_tensors]
+            #Calculate the size of each input tensor
+            dim_sizes = [len(t.size()) for t in in_tensors]
+
+            max_el_ind = els.index(max(els))
+            max_dim_ind = dim_sizes.index(max(dim_sizes))
+
+            if max_el_ind == max_dim_ind:
```

Comment:
This needs to be much more thoroughly documented since a lot of complicated stuff is happening.